### PR TITLE
WIP: Basic Permalink from react-spatial

### PR DESCRIPTION
**Pull request by edellucien**

### CODE REVIEW

File: src/components/Root/Root.js [hunk 1]
```diff
@@ -2,46 +2,114 @@ import React, { PureComponent } from 'react';
...
+    return layers
+      .filter(l => l.isBaseLayer === true)
+      .map(l => l.id)
+      .join(',');
   }
 
   render() {
-    const { title, center } = this.props;
+    const { title, center, zoom, layers, history } = this.props;
+    this.params = {
```
Comment (oterral) [inline src/components/Root/Root.js]: This shoul be in componentDidUpdate

File: src/components/Root/Root.js [hunk 2]
```diff
@@ -2,46 +2,114 @@ import React, { PureComponent } from 'react';
...
+      zoom,
+      x: center[0],
+      y: center[1],
+    };
+
+    const layerNames = this.getLayers(layers);
```
Comment (oterral) [inline src/components/Root/Root.js]: This should be in componentDidUpdate

File: src/components/Root/Root.js [hunk 3]
```diff
@@ -2,46 +2,114 @@ import React, { PureComponent } from 'react';
...
+
+    this.params = {
+      zoom: z || zoom,
+      x: x || center[0],
+      y: y || center[1],
+    };
+
+    const layerNames = this.getLayers();
+    if (layerNames) {
+      this.params.layers = layerNames;
```
Comment (oterral) [inline src/components/Root/Root.js]: Params should be part of ` props` or `state` object`

**Comment by edellucien** [comment]
You can check the changes.

**Review by oterral — APPROVED** [review]
LGTM :v:

**Pull request merged**